src/compute-types/src/time_dependence.rs

```diff
@@ -0,0 +1,221 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+//! Description of how a dataflow follows time, independent of time.
+
+use mz_proto::{RustType, TryFromProtoError};
+use mz_repr::refresh_schedule::RefreshSchedule;
+use mz_repr::Timestamp;
+use proptest::arbitrary::{any, Arbitrary};
+use proptest::prelude::BoxedStrategy;
+use proptest::strategy::Strategy;
+use serde::{Deserialize, Serialize};
+
+include!(concat!(
+    env!("OUT_DIR"),
+    "/mz_compute_types.time_dependence.rs"
+));
+
+/// Description of how a dataflow follows time.
+///
+/// The default value indicates the dataflow follows wall-clock without modifications.
+#[derive(Debug, Clone, Default, Serialize, Deserialize, Eq, PartialEq, Ord, PartialOrd)]
+pub struct TimeDependence {
```

Comment:
I liked the previous implementation more, as it was more explicit. In the new code:
* There is no distinction between an "unset" time_dependence (defaults to None) and a time_dependence that was computed to be "indeterminate" (now None). Perhaps there is another way to distinguish between the two cases when debugging?
* The wall-clock time represented using the default value is an additional thing to keep in mind while paging in the code.